causalml/propensity.py

```diff
@@ -1,46 +1,32 @@
+from abc import ABCMeta, abstractproperty
 import logging
 import numpy as np
 from pygam import LogisticGAM, s
 from sklearn.metrics import roc_auc_score as auc
 from sklearn.linear_model import LogisticRegressionCV
 from sklearn.model_selection import StratifiedKFold, train_test_split
 import xgboost as xgb
-import multiprocessing
 
-logger = logging.getLogger('causalml')
 
+logger = logging.getLogger('causalml')
 
-class ElasticNetPropensityModel(object):
-    """Propensity regression model based on the ElasticNet algorithm.
 
-    Attributes:
-        model (sklearn.linear_model.ElasticNetCV): a propensity model object
-    """
-
-    def __init__(self, n_fold=4, Cs=np.logspace(1e-3, 1 - 1e-3, 4), l1_ratios=np.linspace(1e-3, 1 - 1e-3, 4),
-                 clip_bounds=(1e-3, 1 - 1e-3), cv=None, random_state=None):
-        """Initialize a propensity model object.
+class PropensityModel(metaclass=ABCMeta):
+    def __init__(self, clip_bounds=(1e-3, 1 - 1e-3), **model_kwargs):
+        """
         Args:
-            n_fold (int): the number of cross-validation fold
-            Cs (int or array-like): Each of the values in Cs describes the inverse of regularization strength.
-                If Cs is as an int, then a grid of Cs values are chosen in a logarithmic scale between 1e-4 and 1e4.
-            l1_ratios (array-like): array of l1 ratios (0 <= value <= 1) to iterate through in CV estimator
-            clip_bounds (tuple): lower and upper bounds for clipping propensity scores. Bounds should be implemented
-                such that: 0 < lower < upper < 1, to avoid division by zero in BaseRLearner.fit_predict() step.
-            cv (int or cross-validation generator): The default cross-validation generator used is Stratified K-Folds.
-                If an integer is provided, then it is the number of folds used.
-            random_state (numpy.random.RandomState or int): RandomState or an int seed
 
-        Returns:
-            None
+        clip_bounds (tuple): lower and upper bounds for clipping propensity scores. Bounds should be implemented
+                such that: 0 < lower < upper < 1, to avoid division by zero in BaseRLearner.fit_predict() step.
+        model_kwargs: Keyword arguments to be passed to the underlying classification model.
```

Comment:
Missing indentation. Please align the indentation.

Reply:
Unsure what alignment you're looking for.

`such that` aligned with `lower and ...` above?

Reply:
All three lines (lines 19 ~ 21) should have additional indentation to be one level below than `Args:`.

Reply:
Actually, it's cosmetic, and I'm fine with as-is. Please feel free to merge (squash & merge) this to `main`. Thanks again!

Reply:
Got you. All good. Happy to fix that. Am also a fan of tidy code.